### PR TITLE
refactor: replace char_u

### PR DESCRIPTION
**Pull request by dundargoc**
Work on https://github.com/neovim/neovim/issues/459

### CODE REVIEW

File: src/nvim/indent_c.c [hunk 1]
```diff
@@ -1501,7 +1501,7 @@ static pos_T *find_match_char(char_u c, int ind_maxparen)
   if ((trypos = findmatchlimit(NULL, c, 0, ind_maxp_wk)) != NULL) {
     // check if the ( is in a // comment
     if ((colnr_T)cin_skip2pos(trypos) > trypos->col) {
-      ind_maxp_wk = ind_maxparen - (int)(cursor_save.lnum - trypos->lnum);
+      ind_maxp_wk = ind_maxparen - cursor_save.lnum - trypos->lnum;
```
Comment (zeertzjq) [inline src/nvim/indent_c.c:1504]:
```suggestion
      ind_maxp_wk = ind_maxparen - (cursor_save.lnum - trypos->lnum);
```

File: src/nvim/indent_c.c [hunk 2]
```diff
@@ -1515,7 +1515,7 @@ static pos_T *find_match_char(char_u c, int ind_maxparen)
       trypos = &pos_copy;
       curwin->w_cursor = *trypos;
       if ((trypos_wk = ind_find_start_CORS(NULL)) != NULL) {  // XXX
-        ind_maxp_wk = ind_maxparen - (int)(cursor_save.lnum - trypos_wk->lnum);
+        ind_maxp_wk = ind_maxparen - cursor_save.lnum - trypos_wk->lnum;
```
Comment (zeertzjq) [inline src/nvim/indent_c.c:1518]:
```suggestion
        ind_maxp_wk = ind_maxparen - (cursor_save.lnum - trypos_wk->lnum);
```

Comment (dundargoc) [reply src/nvim/indent_c.c:1518]: oh, right. Math.

File: src/nvim/mbyte.c [hunk 3]
```diff
@@ -1859,7 +1859,7 @@ int mb_tail_off(const char_u *base, const char_u *p)
     }
   }
 
-  if (utf8len_tab[p[-j]] != i + j + 1) {
+  if (utf8len_tab[(uint8_t)p[-j]] != i + j + 1) {
```
Comment (bfredl) [inline src/nvim/mbyte.c:1862]: are the usages of `p[...]` above also safe? we could follow the `p_in` pattern to not need to worry.

Comment (dundargoc) [reply src/nvim/mbyte.c]: I think so. Sure.

**Pull request merged**